[Neeladri Bhuiya] iP

zawedcvg opened the pull request:
# DukeIt
 
 > “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

DukeIt frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it's FREE

Features:

- [X] Managing tasks
- [X] Managing deadlines
- [X] Finding tasks

 If you Java programmer, you can use it to practice Java too. Here's the main method

```java
    public void start(Stage stage) {
        try {
            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
            AnchorPane ap = fxmlLoader.load();
            Scene scene = new Scene(ap);
            stage.setScene(scene);
            fxmlLoader.<MainWindow>getController().setDuke(duke);
            stage.show();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }


```

Commit:
Add Gradle support

zawedcvg committed:
Added basic input and output capabilities

zawedcvg committed:
refactored the code and added listing capabilities

zawedcvg committed:
addded done functionality

zawedcvg committed:
added multiple classes with inheritance

zawedcvg committed:
added semi-automatic UI testing

zawedcvg committed:
addded exceptions and error handling

zawedcvg committed:
added delete

zawedcvg committed:
add local storage and functionality to retrieve data

zawedcvg committed:
add a class to parse date and time

zawedcvg committed:
Merge branch 'level-8'

zawedcvg committed:
refactor code for more OOP

zawedcvg committed:
move everything in Duke and delete .class

zawedcvg committed:
fix import statements

zawedcvg committed:
add testing for some of the files

zawedcvg committed:
add java documentation

zawedcvg committed:
add find functionality

zawedcvg committed:
Merge branch 'branch-level-9'

zawedcvg committed:
coding standard

joelwongjy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,71 @@
+     * @return array containing the sate and time split
+     */
+    public static String[] splitDateAndTime(String description) {
+        return description.split(" ", 2);
```
I really liked that you moved the logic to split the arguments into the parent class. Well done!

joelwongjy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,95 @@
+     */
+
+    public ArrayList<Task> fetch() {
+
```
This could be named clearer. fetchTasks might be a better name?

joelwongjy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,95 @@
+     * @param listOfTasks is the list of tasks that is to be saved in the hard drive
+     */
+    public void save(ArrayList<Task> listOfTasks) {
+        try {
```
Name could be clearer. saveTasks might I suggest?

joelwongjy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+
+    /**
+     * @return string in the format it is saved as
+     */
```
Oops missing method description here. Some of your following methods are also missing the description.

joelwongjy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,107 @@
+
+/** 
+ * The main program where everthing starts
+ *
```
This comment could be clearer. I'd like to know more about what "everything" is 🤔

joelwongjy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+
+    /**
+     * 
+     * @param description the description of the task
```
Do check your javadoc formatting. Should add the period at the end of each line.

joelwongjy reviewed (commented):
LGTM! I really like your abstraction and coding style. Some styling issues but I love the most part of it 😄 Keep up the good work!

zawedcvg committed:
modify .gitignore

zawedcvg committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-gradle

zawedcvg committed:
add gradle support

Benjykoh commented:
Looks great! Clean, well written code :)

zawedcvg committed:
add gui and a new exception

zawedcvg committed:
add assertions

zawedcvg committed:
Merge pull request #2 from zawedcvg/branch-A-Assertions

add assertions

zawedcvg committed:
TaskList class: add ability to enter multiple parameters

Commands mark, unmark and delete can only take in one attribute.

Allowing the user to enter more than one parameter is more user
friendly.

break down the description using split and use these as parameters to
delete/mark/unmark.

zawedcvg committed:
Merge pull request #3 from zawedcvg/A-BCD-Extension

TaskList class: add ability to enter multiple parameters

zawedcvg committed:
fix code quality by changing function names to appropiate ones

zawedcvg committed:
add user guide

zawedcvg committed:
Set theme jekyll-theme-cayman

zawedcvg committed:
Add files via upload

zawedcvg committed:
add fonts

zawedcvg committed:
Merge branch 'master' of https://github.com/zawedcvg/ip
